libretro-common/net/net_http.c

```diff
@@ -373,16 +318,11 @@ bool net_http_connection_iterate(struct http_connection_t *conn)
 
 bool net_http_connection_done(struct http_connection_t *conn)
 {
-   char **location = NULL;
-
    if (!conn)
       return false;
 
-   location     = &conn->location;
-
    if (*conn->scan == '\0')
       return false;
-   *conn->scan  = '\0';
```

Comment:
This was the source of the problem. By nulling out the character here, the check on line 395 was never true.